src/tests/integration/test_generic_db_operations.py

```diff
@@ -1117,3 +1117,74 @@ def test_autocommit_create_table_multiple_cells(
     ).result
 
     assert len(result) == 3
+
+
+@pytest.mark.parametrize(
+    "table, offset, n_rows, expected_rows, expected_columns",
+    [
+        ("test_numbers", 0, 0, [], ["value_one", "value_two"]),
+        ("test_numbers", 5, 0, [], ["value_one", "value_two"]),
+        ("test_numbers", 50, 0, [], ["value_one", "value_two"]),
+        ("test_numbers", 50, 10, [], ["value_one", "value_two"]),
+        ("test_numbers", 0, 2, [(0, 1), (0, 0)], ["value_one", "value_two"]),
+        (
+            "test_numbers",
+            0,
+            10,
+            [
+                (0, 1),
+                (0, 0),
+                (5, 2),
+                (6, 3),
+                (7, 3),
+                (8, 8),
+                (9, 3),
+                (10, 5),
+                (11, 9),
+                (12, 19),
+            ],
+            ["value_one", "value_two"],
+        ),
+        (
+            "test_numbers",
+            0,
+            5,
+            [(0, 1), (0, 0), (5, 2), (6, 3), (7, 3)],
+            ["value_one", "value_two"],
+        ),
+        (
+            "test_numbers",
+            5,
+            5,
+            [(8, 8), (9, 3), (10, 5), (11, 9), (12, 19)],
+            ["value_one", "value_two"],
+        ),
+    ],
+)
+@pytest.mark.parametrize("ip_with_dynamic_db", ALL_DATABASES)
+def test_explore_fetch_sql_with_pagination(
+    ip_with_dynamic_db, request, table, offset, n_rows, expected_rows, expected_columns
+):
+    ip_with_dynamic_db = request.getfixturevalue(ip_with_dynamic_db)
+    ip_with_dynamic_db.run_cell(
+        """
+    %%sql sqlite://
```

Comment:
if you pass sqlite:// you'll overwrite the db and you'll be testing with sqlite instead of the database from the ip_with_dynamic_db, please remove